test/e2e/arktos/multi_tenancy/cmd/framework/testcase.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2020 Authors of Arktos.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+type TestCase struct {
+	Command string `yaml:"Command,omitempty"`
+	TimeOut *int   `yaml:"TimeOut,omitempty"`
+
+	RunBefore string `yaml:"RunBefore,omitempty"`
+
+	RetryCount    *int `yaml:"RetryCount,omitempty"`
+	RetryInterval *int `yaml:"RetryInterval,omitempty"`
+
+	TerminateAfter *int `yaml:"TerminateAfter,omitempty"`
+
+	ShouldFail          bool     `yaml:"ShouldFail,omitempty"`
+	OutputShouldBe      string   `yaml:"OutputShouldBe,omitempty"`
+	OutputShouldContain []string `yaml:"OutputShouldContain,omitempty"`
+
+	OutputShouldNotContain []string `yaml:"OutputShouldNotContain,omitempty"`
+}
+
+func (t *TestCase) Validate(tc *TestConfig) *ErrorList {
+	errList := NewErrorList()
+
+	errList.Add(ErrorIfEmpty("Command", t.Command))
+	errList.Add(ErrorIfNotWithin("RetryCount", t.RetryCount, 0, tc.MaxRetryCount))
+	errList.Add(ErrorIfNotWithin("RetryInterval", t.RetryInterval, 0, tc.MaxRetryInterval))
+	errList.Add(ErrorIfNotWithin("TimeOut", t.TimeOut, 0, tc.MaxTimeOut))
+
+	return errList
+}
+
+func (t *TestCase) Complete(tc *TestConfig) {
```

Comment:
this all this function does is set default values, we could probably rename this function as something like setDefaultValues instead of Complete.

Reply:
This naming is consistent with the naming kubectl code. It is a pretty k8s-style naming.  I will add some comment, though.